File: RoomInn/app/src/main/java/postpc/finalproject/RoomInn/ui/ScanUnityHandler.java
package postpc.finalproject.RoomInn.ui;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.navigation.NavController;
import androidx.navigation.fragment.NavHostFragment;

import com.unity3d.player.UnityPlayer;

import postpc.finalproject.RoomInn.MainActivity;
import postpc.finalproject.RoomInn.R;
import postpc.finalproject.RoomInn.models.RoomInnApplication;

public class ScanUnityHandler extends AppCompatActivity {
    public static Context ctx;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ctx = this;
        setContentView(R.layout.activity_unity_handler);
        Button unityButton = findViewById(R.id.startUnityButton);
        unityButton.setOnClickListener(new View.OnClickListener() {
            public void onClick(View v) {
                Intent thisIntent = getIntent();
                Intent intent = new Intent(ScanUnityHandler.this, ScanUnityPlayerActivity.class);
                intent.putExtra("Scene Index", thisIntent.getStringExtra("Scene Index"));
                startActivity(intent);
            }
        });
    }


    public void backPressedFromRoomScan(){
        // navigate to project page
        super.onBackPressed();
    }

    public void toCallFromUnity(){
        // navigate to windows/ doors fragment

        runOnUiThread(new Runnable() {

            @Override
            public void run() {
                Intent intent = new Intent(ScanUnityHandler.this, MainActivity.class);
                String userId = getIntent().getStringExtra("User ID");
                String roomName =  getIntent().getStringExtra("Room Name");
                intent.putExtra("User ID", userId);
                intent.putExtra("Room Name",roomName);
                startActivity(intent);
//                this.mUnityPlayer.quit();

            }
        });

        NavHostFragment navHostFragment = (NavHostFragment) getSupportFragmentManager().findFragmentById(R.id.nav_host_fragment);
        if (navHostFragment != null){
            NavController navController = navHostFragment.getNavController();
            navController.navigate(R.id.action_profileFragment2_to_floorPlanPlacingFragment);
        }
    }
}

File: RoomInn/app/src/main/java/postpc/finalproject/RoomInn/ui/ScanUnityHandler.kt
package postpc.finalproject.RoomInn.ui

import android.content.Intent
import android.content.pm.ActivityInfo
import android.os.Bundle
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity
import postpc.finalproject.RoomInn.R
import kotlin.properties.Delegates

class ScanUnityHandler : AppCompatActivity() {
    private lateinit var userId: String
    private lateinit var thisIntent: Intent
    private var returnTo by Delegates.notNull<Int>()

    public override fun onCreate(savedInstanceState: Bundle?) {
        //sets the activity to free orientation
        requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_USER

        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_unity_handler)
        val unityButton = findViewById<Button>(R.id.startUnityButton)
        unityButton.setOnClickListener {
            thisIntent = intent
            userId = thisIntent.getStringExtra("User ID")!!
            returnTo = thisIntent.getIntExtra("Return To",0)
            val intent = Intent(this@ScanUnityHandler, ScanUnityPlayerActivity::class.java)
            intent.putExtra("User ID", userId)
            intent.putExtra("Return To", returnTo)
            startActivity(intent)
        }
    }
}